pkg/build/builder/common.go

```diff
@@ -66,6 +66,25 @@ func buildInfo(build *api.Build) []KeyValue {
 	return kv
 }
 
+// updateBuildStatus updates the current status of the build only if there was
+// an error
+func updateBuildStatus(c client.BuildInterface, build *api.Build) {
```

Comment:
all the more reason that this function should not actually make the update api call (nor should updateRevision), but rather let the caller do so (and decide how to handle a failure to update).